blog/en/blog/2025/03/31/march-monthly-report.md

```diff
@@ -0,0 +1,164 @@
+
+## Feature Highlights
+
+### 1. Add `ai-prompt-guard` Plugin
+
+PR: https://github.com/apache/apisix/pull/12008
+
+Contributor: [Revolyssup](https://github.com/Revolyssup)
+
+The `ai-prompt-guard` plugin safeguards your AI endpoints by inspecting and validating incoming prompt messages. It checks the content of requests against user-defined allowed and denied patterns to ensure that only approved inputs are processed. Based on its configuration, the plugin can either examine just the latest message or the entire conversation history, and it can be set to check prompts from all roles or only from end users.
+
+### 2. Add `ai-rate-limiting` Plugin
+
+PR: https://github.com/apache/apisix/pull/12037
+
+Contributor: [shreemaan-abhishek](https://github.com/shreemaan-abhishek)
+
+The `ai-rate-limiting` plugin enforces token-based rate limiting for requests sent to LLM services. It helps manage API usage by controlling the number of tokens consumed within a specified time frame, ensuring fair resource allocation and preventing excessive load on the service. It is often used with `ai-proxy-multi` plugin.
+
+### 3. Add `ai-request-rewrite` Plugin
+
+PR: https://github.com/apache/apisix/pull/12036
+
+Contributor: [LiteSun](https://github.com/LiteSun)
+
+The `ai-request-rewrite` plugin leverages predefined prompts and AI services to intelligently modify client requests, enabling AI-powered content transformation before forwarding to upstream services.
+
+### 4. Support Proxying OpenAI-Compatible LLMs
+
+PR: https://github.com/apache/apisix/pull/12004
+
+Contributor: [shreemaan-abhishek](https://github.com/shreemaan-abhishek)
+
+This PR introduces a new provider, `openai-compatible`, that can proxy requests to OpenAI-compatible LLMs.
+
+### 5. Store JWT in the Request Context in `jwt-auth` Plugin
+
+PR: https://github.com/apache/apisix/pull/11675
+
+Contributor: [mikyll](https://github.com/mikyll)
+
+The `jwt-auth` plugin now has a new parameter: `store_in_ctx`. It allows the configuration of When enabled (default is off), it stores the validated JWT object in the request context.
+
+This feature is especially beneficial for two reasons:
+
+- The JWT can be removed from request attributes when `hide_credential = true`, this offers a secure alternative for token passing without exposure.
+
+- It prevents custom plugins from duplicated code, for example, retrieve and parse JWT objects.
+
+### 6. Add JWT Audience Validator in `openid-connect` Plugin
+
+PR: https://github.com/apache/apisix/pull/11987
+
+Contributor: [bzp2010](https://github.com/bzp2010)
+
+Add JWT audience authentication to the `openid-connect` plugin to:
+
+- Allow the configuration of audience claim to enforce validation of JWT Audience Validator.
+- Asserts that it should be equal to or contain the `client_id` to comply with the OIDC specification requirements; otherwise, rejects the request.
```

Comment:
```suggestion
- Asserts that it should be equal to or contain the `client_id`  when the `client_id` is a string and an array respectively, to comply with the OIDC specification requirements; otherwise, rejects the request.
```